src/xiana/interceptor.clj

```diff
@@ -143,42 +145,36 @@
               (update state :request dissoc :body :body-params)
               state))})
 
-(defn- valid? [?schema data]
+(defn valid? [?schema data]
   (let [value (m/decode ?schema data (mt/transformer
                                        (mt/json-transformer)
                                        (mt/string-transformer)
                                        (mt/strip-extra-keys-transformer)))
         details (m/explain ?schema value)]
     (if (nil? details)
       value
-      (throw (ex-info "Request schema validation error"
-                      {:xiana/response {:status 400
-                                        :body   (pr-str details)}})))))
+      (throw (ex-info "Request schema validation/coercion error"
+                      {:xiana/response {:details (me/humanize details)}})))))
 
 (def coercion
   "On enter: validates request parameters
   On leave: validates response body
   on request error: responds {:status 400, :body \"Request coercion failed\"}
   on response error: responds {:status 400, :body \"Response validation failed\"}"
   {:enter (fn [state]
-            (if (= :options (-> state :request :request-method))
+            (if (= :options (get-in state [:request :request-method]))
               state
               (let [path (get-in state [:request-data :match :path-params])
                     query (get-in state [:request :query-params])
                     form-params (or (not-empty (get-in state [:request :form-params]))
                                     (not-empty (get-in state [:request :multipart-params]))
                                     (not-empty (get-in state [:request :body-params])))
                     method (get-in state [:request :request-method])
-                    schemas (merge (get-in state [:request-data :match :data :parameters])
-                                   (get-in state [:request-data :match :data method :parameters]))
-                    cc (cond-> {}
-                         (:path schemas)
-                         (assoc :path (valid? (:path schemas) path))
-
-                         (:query schemas)
-                         (assoc :query (valid? (:query schemas) query))
-
-                         (:form schemas)
-                         (assoc :form (valid? (:form schemas) form-params)))]
+                    schemas (or (get-in state [:request-data :match :data method :parameters])
+                                (get-in state [:request-data :match :data :parameters]))
```

Comment:
I'd keep the merge as it was, to be able to define the path related parameters once, and method parameters separately.

Reply:
Ok.